source/grid/grid_generator.cc

```diff
@@ -4266,20 +4266,16 @@ namespace GridGenerator
     // compute cells to remove
     std::set<typename Triangulation<dim, spacedim>::active_cell_iterator>
       cells_to_remove;
-    std::copy_if(
-      rectangle.active_cell_iterators().begin(),
-      rectangle.active_cell_iterators().end(),
-      std::inserter(cells_to_remove, cells_to_remove.end()),
-      [&](
-        const typename Triangulation<dim, spacedim>::active_cell_iterator &cell)
-        -> bool {
+    for (const auto &cell : rectangle.active_cell_iterators())
+      {
+        bool remove_cell = true;
         for (unsigned int d = 0; d < dim; ++d)
```

Comment:
Should we keep the shortcut?
```suggestion
        for (unsigned int d = 0; d < dim && remove_cell; ++d)
```